### PR TITLE
Fix enum sequence and struct sequence validation

### PR DESCRIPTION
**Pull request by jonathan-r-thorpe**
- Enum sequences and struct sequences are failing the constraint validation even though they were valid.
- This is because the enum and struct sequences were not being treated like arrays.  
- This PR fixes the problem. 
- It also includes improved logging/error reporting.
- The draft MS-05-02 test suite in NMOS Testing has been updated to trap this bug.

### CODE REVIEW

**Review by lo-simon — CHANGES_REQUESTED** [review]
Suggest minor changes. Rest LGTM.

File: Development/nmos/control_protocol_methods.cpp [hunk 1]
```diff
@@ -30,6 +30,7 @@ namespace nmos
         // unknown property
         utility::stringstream_t ss;
         ss << U("unknown property: ") << property_id.serialize() << U(" to do Get");
+        slog::log<slog::severities::error>(gate, SLOG_FLF) << utility::conversions::to_utf8string(ss.str());
```
[the remark this answers is not in the document]
Comment (garethsb) [reply Development/nmos/control_protocol_methods.cpp:33]: Why write into a separate stringstream first at all?

File: Development/nmos/control_protocol_methods.cpp [hunk 2]
```diff
@@ -266,15 +275,17 @@ namespace nmos
             }
             catch (const nmos::control_protocol_exception& e)
             {
-                slog::log<slog::severities::error>(gate, SLOG_FLF) << "Add sequence item: " << property_id.serialize() << " value: " << val.serialize() << " error: " << e.what();
-
+                utility::stringstream_t ss;
+                ss << "Add sequence item: " << property_id.serialize() << " value: " << val.serialize() << " error: " << e.what();
+                slog::log<slog::severities::error>(gate, SLOG_FLF) << utility::conversions::to_utf8string(ss.str());
```
[the remark this answers is not in the document]
Comment (garethsb) [reply Development/nmos/control_protocol_methods.cpp:280]: Why is the `ss.str()` not used in the following `return details::make_nc_method_result(...)` ?

[the remark this answers is not in the document]
Comment (jonathan-r-thorpe) [reply Development/nmos/control_protocol_methods.cpp:280]: Good spot - it should be:
`return details::make_nc_method_result_error({ nc_method_status::parameter_error}, ss.str())`

**Pull request merged**